enhancements/sig-architecture/81-addon-lifecycle/kustomize-rollout-examples/02-rollout-strategy/clustermanagement-addon.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: cool-cluster-scoped-addon-configuration
+spec:
+  rolloutStrategy:
+    strategy: RollingUpdateWithCanary
+    rollingUpdateWithCanary:
+      # clusters in this placement will receive any update first.  The ManagedClusterAddOn must be Available
+      # and not Degraded and have the LastObservedGeneration indicating it saw our update.
+      # I'm not completely certain the LastObservedGeneration semantic covers, "I have applied this configuration",
+      # but that's the goal.
+      canaryPlacement: canary-placement
+      maxConcurrentlyUpdating: 10%
+  # unless there is rule about how to handle multiple placements covering the same managed cluster, there is logically
+  # only a single configuration possible per ClusterManagementAddOn
+  # This field will be substituted by kustomize in this example, so it will have a hash and that will start a canary
+  # rollout.
+  hubScopedAddOnConfigName: standard-config
```

Comment:
I tried to document here how I envisioned this working.  This field works very well with kustomize by having SHA suffix substitution cause a controlled rollout with a canary on desired changes.  Which can include version updates.